### PR TITLE
Refactor TestMoonlight

### PR DESCRIPTION
**Pull request by mc-anton-karpov**
## Context

There is a small refactoring of `TestMoonlight` in terms of fixing some issues and reusing directly `Moonlight`'s logic.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation including README.md
- [x] My changes generate no new warnings
- [x] My changes generate no breaking changes

### CODE REVIEW

**Review by kuldar-daniel — APPROVED** [review]
Looks a lot better

File: Sources/MoonlightTests/MoonlightTestCase.swift [hunk 1]
```diff
@@ -37,15 +38,15 @@ public extension MoonlightTestCase {
...
-            .last()
+        Moonlight.start(
+            initialState: initialState,
+            environment: environment,
+            feedback: { _ in events.publisher.eraseToAnyPublisher() },
+            transform: transform,
+            apply: apply,
+            store: { cancellables.update(with: $0) }
+        )
+            .first()
```
Comment (kuldar-daniel) [inline Sources/MoonlightTests/MoonlightTestCase.swift:49]: why first?

Comment (mc-anton-karpov) [reply Sources/MoonlightTests/MoonlightTestCase.swift:49]: for cancelling the subscription after getting the result

Comment (kuldar-daniel) [reply Sources/MoonlightTests/MoonlightTestCase.swift:49]: what if we pass it a list of events that will mutate the state two times, will the second one be ignored?

Comment (mc-anton-karpov) [reply Sources/MoonlightTests/MoonlightTestCase.swift:49]: no, it will be the result after all pasted events

**Review by mc-evgeny-zborovsky — APPROVED** [review]

**Review by mc-aleksander-stos — APPROVED** [review]

**Pull request merged**